RFC/src/RFC-0170_NetworkCommunicationProtocol.md

```diff
@@ -0,0 +1,302 @@
+|---             |---                      |---                 |
+| Validator Node | Consensus Node          | Registration       |
+| Base Node      | Consensus Node          | Derived            |
+| Wallet         | Consensus Client        | Derived            |
+| Token Wallet   | Consensus Client        | Derived            |
+
+Note that [Mining Server]s and [Mining Worker]s are excluded from the Tari communication network.
+A Mining Worker will have a local or remote P2P connection with a Mining Server.
+A Mining Server will have a local or remote P2P connection with a Base Node.
+They do not need to make use of the communication network and they are not responsible for propagating any messages on the network.
+The parent Base Node will perform any communication tasks on the Tari communication network on their behalf.
+
+#### Online Communication Address, Peer Address and Routing Table
+
+Each CC and CN on the Tari communication network will have identification cryptographic keys, a node id and an online communication address.
+The online communication address can be either an IPv4, IPv6, URL, Tor(Base32) or I2P(Base32) address and can be stored using the network address type as follows:
+
+| Description  | Data type  | Comments                                     |
+|---           |---         |---                                           |
+| address type | uint4      | Specify if IPv4/IPv6/Url/Tor/I2P             |
+| address      | char array | IPv4, IPv6, Tor(Base32), I2P(Base32) address |
+| port         | uint16     | port number                                  |
+
+The address type is used to determine how to interpret the address characters. An I2P address can be interpreted as "{52 address characters}.b32.i2p". The Tor address should be interpreted as "http://{16 or 52 address chars}.onion/". The IPv4 and IPv6 address can be stored in the address field without modification. URL addresses can be used for nodes with dynamic IP addresses. 
+
+Each CC or CN has a local routing table that contains the online communication addresses of all CCs and CNs on the Tari communication network known to that CC or CN.
+When a CC or CN wants to join the Tari communication network, the online communication address of at least one other CN that is part of the network needs to be known.
+The online communication address of the initial CN can either be manually provided or a bootstrapped list of "reliable" and persistent CNs can be provided with the Validator Node, Base Node, Wallet or Token Wallet software.
+The new CC or CN can then request additional peer contact information of other CNs from the initial peers to extend their own routing table.
+
+The routing table consists of a list of peer addresses that link node ids, public identification keys and online communication addresses of each known CC and CN.
+
+The Peer Address stored in the routing table can be implemented as follows:
```

Comment:
I think this table should be added to the MAY requirement category.

Reply:
Changed to "MAY"